plugins/scale/scale.cpp

```diff
@@ -691,32 +692,59 @@ class wayfire_scale : public wf::plugin_interface_t
         output->render->damage_whole();
     }
 
-    /* Returns a list of views for all workspaces */
+    /* Returns a list of views for all workspaces
+     * Also takes minimized views out of the minimized layer
+     * and stores the state */
     std::vector<wayfire_view> get_all_workspace_views()
     {
         std::vector<wayfire_view> views;
 
+        uint32_t layer_mask = wf::LAYER_WORKSPACE;
+        if (show_minimized)
+        {
+            layer_mask |= wf::LAYER_MINIMIZED;
+        }
+
         for (auto& view :
-             output->workspace->get_views_in_layer(wf::LAYER_WORKSPACE))
+             output->workspace->get_views_in_layer(layer_mask))
         {
             if ((view->role != wf::VIEW_ROLE_TOPLEVEL) || !view->is_mapped())
             {
                 continue;
             }
 
+            if (show_minimized)
+            {
+                if (output->workspace->get_view_layer(view) == wf::LAYER_MINIMIZED)
+                {
+                    output->workspace->add_view(view, wf::LAYER_WORKSPACE);
+                    view->store_data(
```

Comment:
it would have been possible to store this as a vector, but I think storing custom data is always better since any plugin or core can always have information about custom view state

Reply:
although on second thought, a vector loop would only iterate over these particular views.